consul/catalog_endpoint.go

```diff
@@ -36,6 +37,17 @@ func (c *Catalog) Register(args *structs.RegisterRequest, reply *struct{}) error
 			return fmt.Errorf("Must provide service name with ID")
 		}
 
+		// Verify that the service name is valid per RFC-1123 and
+		// RFC-2782.
+		//
+		// FIXME(sean@): After 0.7 has been branched, change from a
+		// warning to an error for the 0.8 release.
+		re := regexp.MustCompile(`^(?i:[a-z0-9]|[a-z0-9][a-z0-9\-]{0,61}[a-z0-9])$`)
```

Comment:
I think you can just do this as a package-global var. The `MustCompile` is little odd during runtime since it can panic. I'd probably just make it package-global to keep that out of the code path.